### PR TITLE
Add unit test for reading Parquet by column name

### PR DESCRIPTION
**Pull request by cawallin**

### CODE REVIEW

**Comment by cawallin** [comment]
see https://github.com/facebook/presto/pull/2258 for some background

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java [hunk 1]
```diff
@@ -277,6 +278,40 @@ public boolean apply(TestColumn testColumn)
...
+
+        HiveOutputFormat<?, ?> outputFormat = new MapredParquetOutputFormat();
+        InputFormat<?, ?> inputFormat = new MapredParquetInputFormat();
+        @SuppressWarnings("deprecation")
+        SerDe serde = new ParquetHiveSerDe();
+        File file = File.createTempFile("presto_test", "parquet");
+        file.delete();
+        try {
+            // Reverse the order of the columns to test access by name, not by index
+            Collections.reverse(testColumns);
```
Comment (rschlussel-zz) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java]: shouldn't it be enough to  reverse the columns only once (i.e. only when you search through)?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java [hunk 2]
```diff
@@ -277,6 +278,40 @@ public boolean apply(TestColumn testColumn)
...
+                                PrimitiveCategory.DATE,
+                                PrimitiveCategory.BINARY))
+                        .collect(Collectors.toList());
+
+        HiveOutputFormat<?, ?> outputFormat = new MapredParquetOutputFormat();
+        InputFormat<?, ?> inputFormat = new MapredParquetInputFormat();
+        @SuppressWarnings("deprecation")
+        SerDe serde = new ParquetHiveSerDe();
+        File file = File.createTempFile("presto_test", "parquet");
+        file.delete();
```
Comment (rschlussel-zz) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java:276]: why does this get deleted twice?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java (hunk 1 above)
Comment (cawallin) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java]: hm. I suppose that's ok, it'll just have slightly different results than all of the other tests

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java (hunk 2 above)
Comment (cawallin) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java:276]: This first deleting is just to reserve the filename I think (then on line 303 we write out a new parquet file)

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java [hunk 3]
```diff
@@ -285,37 +320,37 @@ public void testParquetThrift()
...
-                    )
-                ),
-                null,
-                arrayBlockOf(personType,
-                        rowBlockOf(ImmutableList.of(nameType, BIGINT, VARCHAR, new ArrayType(phoneType)),
-                                rowBlockOf(ImmutableList.of(VARCHAR, VARCHAR), "Bob", "Roberts"),
-                                0,
-                                "bob.roberts@example.com",
-                                arrayBlockOf(phoneType, rowBlockOf(ImmutableList.of(VARCHAR, VARCHAR), "1234567890", null))
+                new TestColumn(
```
Comment (cawallin) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java]: This is just a formatting change -- I guess I should probably remove it.

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java (hunk 2 above)
Comment (rschlussel-zz) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveFileFormats.java:276]: that seems like a very strange way of doing things, but I guess it's fine if the other tests are doing the same.

**Comment by rschlussel-zz** [comment]
lgtm

**Pull request merged**